DCA/adapter_base.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2022, Ilya Syresenkov, Kirill Ivanov and Anastasiia Kornilova
```

Comment:
Adapter base is common for all methods, let's move to separate external folder `common` as in Kirill examples